ginger-trout: even more styling fixes

ttseng opened the pull request:
## Links
* Remix link: https://ginger-trout.glitch.me/
* Manuscript link: 
https://glitch.manuscript.com/f/cases/3328544/Adjust-button-margins-underneath-embed
https://glitch.manuscript.com/f/cases/3328543/Adjust-bottom-margin-for-ProjectsContainer-to-add-spacing-between-container-report-abuse-button
https://glitch.manuscript.com/f/cases/3328541/Segmented-Button-Margins-are-off
https://glitch.manuscript.com/f/cases/3328542/Need-to-handle-overflow-of-markdown-in-project-descriptions

## Changes:
* Fixed overflow issue with markdown in project descriptions:
https://ginger-trout.glitch.me/search?q=saskia
https://glitch.com/search?q=saskia
![image](https://user-images.githubusercontent.com/519336/57795428-edd8c080-7713-11e9-9b98-ea5fa91daf7d.png)

* Fixed segmented button styling (note that there is an override in the add-project-to-collection-pop)
![image](https://user-images.githubusercontent.com/519336/57795468-12cd3380-7714-11e9-9b80-9e9044541ae9.png)

* Fixed margins for projects-list__projectsGrid, which was affecting the placement of the Report Abuse button (note there are now overrides when the projects-list is within a container, like more by or recent projects)
https://ginger-trout.glitch.me/@scientiffic-test
![image](https://user-images.githubusercontent.com/519336/57795613-6c356280-7714-11e9-9373-77151700af99.png)

* Adjusted collection-options-pop placement
![image](https://user-images.githubusercontent.com/519336/57795694-9a1aa700-7714-11e9-92b2-aea1a52446f5.png)

* Edited small button padding in response to https://glitch.manuscript.com/f/cases/3328544/Adjust-button-margins-underneath-embed

* Fixed grid size for collection pages' projectsGridCompact (currently showing a max of 2 columns on prod instead of 3)
![image](https://user-images.githubusercontent.com/519336/57795946-38a70800-7715-11e9-860e-7e8b70f95cde.png)


## How To Test:
* Confirm the changes listed above and browse to see if the changes might have affected anything else unintentionally?

modernserf commented on src/components/containers/projects-list.styl:
```diff
@@ -1,16 +1,13 @@
 .projectsGridCompact
-  --row-gap: 20px
-  --min-width: 250px
+  --row-gap: 20px !important
```
What do these do? I'm not seeing any difference between these `!important`s being present or not

modernserf replied:
oh, i see -- this is a specificity thing. hmm, i'd like to fix this in the grid container rather than using !important here

ttseng replied:
yeah, I was trying to figure that out because on staging, it seems like css from `projects-list.styl` overrides anything from `grid-styl`:
![image](https://user-images.githubusercontent.com/519336/57797187-e1566700-7717-11e9-921c-9333b7bf6746.png)

But on production the opposite is true...
![image](https://user-images.githubusercontent.com/519336/57797226-f3d0a080-7717-11e9-8516-105656c39a77.png)

modernserf replied:
oh, how about in `containers/grid.styl`

```
.grid
  display: grid  
  grid-row-gap: var(--row-gap, 10px)
  grid-column-gap: var(--column-gap, 10px)
  grid-template-columns: repeat(auto-fill, minmax(var(--min-width, 300px), 1fr))
  list-style-type: none
  margin: 0
  padding: 0
```

ttseng replied:
yeah! that works perfectly

sarahzinger commented on src/components/collections-list/index.js:
```diff
@@ -97,9 +97,7 @@ function CollectionsList({ collections: rawCollections, title, isAuthorized, may
-          <div className={styles.createCollectionButtonWrap}>
-            <CreateCollectionButton maybeTeam={maybeTeam} />
-          </div>
+          <CreateCollectionButton maybeTeam={maybeTeam} />
```
I feel like the margin here though makes sense since it's internal to the component you know? it's just the external ones that I think are potentially wonky...or at least that's my interpretation

ttseng replied:
The reason I got rid of the `createCollectionButtonWrap` is because it was adding an unneeded bottom margin to the `Create Collection button (compare changes on ginger-trout on left with what's on production on right)

![image](https://user-images.githubusercontent.com/519336/57799198-85421180-771c-11e9-8588-60c1a2cec49a.png)

(and the distance between the button and the collection item already feels further than normal because of the lack of user avatars)

sarahzinger replied:
oh sorry I wasn't being clear I was agreeing with your change here of not wrapping if that makes sense!

sarahzinger commented on styles/pop-overs.styl:
```diff
@@ -301,6 +298,8 @@ popover-secondary-background()
       width: 15px
       border-radius: 3px
+  .segmented-button-wrap
+    margin-bottom: -1rem
```
I think rather than wrapping when we need to negate a margin, I think it might be easier to maintain to not have a margin in the core component, and only add it in a wrapper when we need it. That seems to be the convention we're suggesting in our docs anyway https://github.com/FogCreek/Glitch-Community/blob/master/STYLE.md#components this way the components don't know where they are being used on the page and the parent components don't need to know the internals of their components to reset their margins.

ttseng commented:
one other small adjustment with user-options-pop dropdown - added some left margin between the dropdown and the avatar 
![image](https://user-images.githubusercontent.com/519336/57801242-50848900-7721-11e9-95b8-d4958ec3eb7d.png)

ttseng commented:
also fixed collection avatar SVG sizing in firefox by adding `min-width:0` to autocomplete avatarContainer (https://github.com/philipwalton/flexbugs/issues/184)
![image](https://user-images.githubusercontent.com/519336/57864401-d0662e00-77c9-11e9-9e08-c47035ce688e.png)

and removed the "anonymous projects expire after 2 weeks" message on the homepage for anonymous users with projects (since this is no longer the case)
![image](https://user-images.githubusercontent.com/519336/57864716-64d09080-77ca-11e9-93b9-f42d156636e8.png)

not planning on making any additional changes, so this is ready for any additional comments / approval @modernserf @sarahzinger!
